.github/workflows/build.yml

```diff
@@ -1,6 +1,10 @@
 name: Haskell CI
 
-on: [push]
+on:
+  push:
+    branches: ['*']
```

Comment:
Ah, I think the issue with the Hydra timeout is that this trigger does not fire on commits on `bors/staging`, as the `*` glob does not match the `/` (see [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)), so it probably has to be included in addition.

Reply:
Oh, so it means if one pushes a branch `coot/feature-x` it will not trigger as well.

Reply:
Yes, that is my understanding, one would have to use branch names without `/` when one wants to run CI without creating a PR.

Reply:
I checked on another repo that at least when when one pushes to a branch with open PR, then just `pull_request` is enough to trigger CI.  I previously thought that this is only done when `push` is set as well.

I think it's ok to be explicit in `push` and allow all pull request, e.g.

```yaml
on:
  push:
    branches:
      - 'bors/*'
  pull_request:
```

It seems the ledger team is not really using `bors` for merging, that's why it wasn't an issue for them.

Reply:
Now should not trigger the same workflow even for PRs from forks.